### DISCUSSION TITLE
Reading version string from a file?

### DISCUSSION THREAD

**Discussion by kwaegel** [mtkennerly/poetry-dynamic-versioning#177] [Q&A] [answered]
This might be out of scope for this plugin, but is there a way to read a version string from a file? This doesn't seem to be built into Poetry itself, and this is the most popular dynamic versioning plugin.

Something similar to this feature in PDM:
https://backend.pdm-project.org/metadata/#read-from-a-static-string-in-the-given-file-path

One potential workaround might be to set `POETRY_DYNAMIC_VERSIONING_BYPASS=$(cat version.txt)`, but that feels kind of fragile.

**Comment by mtkennerly** [comment, marked as answer]
Hi! On one hand, I feel like this isn't exactly in scope, but on the other hand, it's a simple enough feature and the scope has gradually expanded a bit since the project started, so I don't really mind adding this if it helps :)

This will be included in the next release in a `from-file` config section: https://github.com/mtkennerly/poetry-dynamic-versioning/commit/3890c4ea467635d685350f32042fb49b3f2cda78

**Reply by kwaegel** [reply]
Thanks! I'm dealing with a couple of projects that currently have a somewhat complicated system for generating the version, and just reading the existing value from a file will make it a lot easier to migrate them over to using Poetry without breaking something.
